### PR TITLE
Add monitor struct/handler

### LINKED ISSUE #464: reduce monitor overhead
Currently, monitoring via the `/__monitor__` endpoint scales linearly with <# of monitor requests> and <# of signers>.

This:

* consumes CPU and memory representing a potential DoS vector 
* introduces latency for real signing requests especially for signers that shell out or lock and shell out like the `gpg2` signer (TODO: measure actual CPU and memory overhead)
* adds complexity to deployments i.e. what succeeds with one config might not work with another

### PR DESCRIPTION
**Pull request by gterzian**
FIX #464

### CODE REVIEW

**Comment by gterzian** [comment]
```
--- FAIL: TestVerifyCOSESignaturesErrs (0.22s)
    cose_test.go:783: verifyCOSESignatures case 9 returned 'xpi: failed to verify EECert 0: x509: certificate has expired or is not yet valid'
```

Is this one expected to fail?

**Comment by gterzian** [comment]
@g-k r?

Looks like that one unit-test keeps failing...

**Comment by g-k** [comment]
Yeah, that test cert might have expired. Thanks for the PR!

**Comment by g-k** [comment]
Meaning to get to this, I want to fix unit tests in CI first.

File: monitor_handler.go [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
-	"go.mozilla.org/autograph/signer"
 )
 
 const monitorAuthID = "monitor"
 
 // MonitoringInputData is the data signed by the monitoring handler
 var MonitoringInputData = []byte(`AUTOGRAPH MONITORING`)
 
-func (a *autographer) handleMonitor(w http.ResponseWriter, r *http.Request) {
+func (m *monitor) handleMonitor(w http.ResponseWriter, r *http.Request) {
```
Comment (g-k) [inline monitor_handler.go:16]: :+1: this makes a lot more sense than having it on `autographer`

File: tools/autograph-monitor/go.sum [hunk 2]
```diff
@@ -38,6 +38,8 @@ github.com/DataDog/datadog-go v2.3.0+incompatible h1:ypCT7tHzMUrUKjdHX6GGEsGq0yi
 github.com/DataDog/datadog-go v2.3.0+incompatible/go.mod h1:LButxg5PwREeZtORoXG3tL4fMGNddJ+vMq1mwgfaqoQ=
 github.com/DataDog/datadog-go v3.4.0+incompatible h1:LZ0OTmlvhCBT0VYUvhGu8Lrc7WqNCj6Zw9HnMi0V6mA=
 github.com/DataDog/datadog-go v3.4.0+incompatible/go.mod h1:LButxg5PwREeZtORoXG3tL4fMGNddJ+vMq1mwgfaqoQ=
+github.com/DataDog/datadog-go v3.7.2+incompatible h1:o4QtYjBU/rG58VPh8Ne6F65YiMY5/v5q4WdY/HvRYMQ=
```
Comment (g-k) [inline tools/autograph-monitor/go.sum:41]: If these dependency updates aren't required for the feature we should drop them.

Comment (g-k) [reply tools/autograph-monitor/go.sum:41]: oh wait nevermind I missed that this was the tools/ monitor module.

**Comment by gterzian** [comment]
@g-k Thanks, comments addressed and rebased, let's see what the unit-tests do...

**Comment by gterzian** [comment]
@g-k Are those failures infra related or the changes actually broke something? I can't tell...

**Comment by g-k** [comment]
Unit tests will always fail, because the API key to publish coverage info
is only injected for non-fork or admin-initiated CI builds.

Not sure what's going on with the app hsm monitor failure, but I'll look at
that too. I don't think it's related to these changes since the regular
monitor run passes.

I'm off for awhile, but I'll look at this again later next week.

File: main.go [hunk 3]
```diff
@@ -284,6 +291,7 @@ func newAutographer(cachesize int) (a *autographer) {
 	a = new(autographer)
 	a.authBackend = newInMemoryAuthBackend()
 	a.nonces, err = lru.New(cachesize)
+	a.exit = make(chan interface{})
```
[the remark this answers is not in the document]
Comment (gterzian) [reply main.go:297]: I added a comment on the `autographer` struct for that field.

[the remark this answers is not in the document]
Comment (g-k) [reply main.go:297]: Thanks!

**Comment by g-k** [comment]
Repro'd the `error: Get http://autograph-app-hsm:8001/__monitor__: dial tcp: lookup autograph-app-hsm on 127.0.0.11:53: no such host` error locally.

**Comment by g-k** [comment]
Rebased and landed in #522. Thank you!

**Pull request closed without merging**